src/System.Collections/src/System/Collections/Generic/SortedSet.cs

```diff
@@ -1796,19 +1781,67 @@ public Node(T item, bool isRed = true)
             public Node Right { get; set; }
 
             public bool IsRed { get; set; }
+
+            public Node DeepClone(int count)
+            {
+                Debug.Assert(count == GetCount());
+                
+                // Breadth-first traversal to recreate nodes, preorder traversal to replicate nodes.
+
+                var originalNodes = new Stack<Node>(2 * Log2(count) + 2);
+                var newNodes = new Stack<Node>(2 * Log2(count) + 2);
+                Node newRoot = ShallowClone();
+
+                Node originalCurrent = this;
+                Node newCurrent = newRoot;
+
+                while (originalCurrent != null)
+                {
+                    originalNodes.Push(originalCurrent);
+                    newNodes.Push(newCurrent);
+                    newCurrent.Left = originalCurrent.Left?.ShallowClone();
+                    originalCurrent = originalCurrent.Left;
+                    newCurrent = newCurrent.Left;
+                }
+
+                while (originalNodes.Count > 0)
+                {
+                    originalCurrent = originalNodes.Pop();
+                    newCurrent = newNodes.Pop();
+
+                    Node originalRight = originalCurrent.Right;
+                    Node newRight = originalRight?.ShallowClone();
+                    newCurrent.Right = newRight;
+
+                    while (originalRight != null)
+                    {
+                        originalNodes.Push(originalRight);
+                        newNodes.Push(newRight);
+                        newRight.Left = originalRight.Left?.ShallowClone();
+                        originalRight = originalRight.Left;
+                        newRight = newRight.Left;
+                    }
+                }
+
+                return newRoot;
+            }
+
+            public int GetCount() => 1 + (Left?.GetCount() ?? 0) + (Right?.GetCount() ?? 0);
```

Comment:
@stephentoub I can't do a `[Conditional("DEBUG")]` here because the return type is non-void, but I'll use an `#if` and throw if this method is called in Release.